app/search/gleaner.py

```diff
@@ -228,6 +249,9 @@ def convert_result(self, sparql_result_dict):
             result['urls'].append(url)
         if sameAs is not None:
             result['urls'].append(sameAs)
+        if validators.url(result['id']):
```

Comment:
I realized that since CCHDO datasets don't have a url given in their metadata, we could substitute the one that is in their ID for the url, so we still have a link back to the dataset landing page as the result title when they get displayed to the user.